### PR TITLE
par2 0.8.0 (adds OpenMP support)

### PR DESCRIPTION
**Pull request by outlyer**
* Upgrade to new upstream version
* Optional support for new higher performance OpenMP multi-threading via LLVM build

- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

File: Formula/par2.rb [hunk 1]
```diff
@@ -1,18 +1,19 @@
...
-  sha256 "e602db3d8bdc49e2cb9e0e089ec31cd262e661ef7450f5d556e43a97a299e71d"
+  url "https://github.com/Parchive/par2cmdline/releases/download/v0.8.0/par2cmdline-0.8.0.tar.bz2"
+  sha256 "496430e185f2d82e54245a0554341a1826f06c5e673fa12a10f176c7f9b42964"
 
-  bottle do
-    cellar :any_skip_relocation
-    sha256 "a825515ff251975d362998560f5e2e046ce3a9f753be106bbf3717c6d411b7fb" => :high_sierra
-    sha256 "6c3432e8a1b7e8ceeabb380af04d13123c9fb542fab7caf62fe8201f3d1adee2" => :sierra
-    sha256 "7257b39640dcf1894c2329129406a573b04dd263bf3b06283dc854b5ed17cf8e" => :el_capitan
-  end
```
Comment (fxcoudert) [inline Formula/par2.rb]: Do not touch the bottle block. It is automatically updated upon merging.

File: Formula/par2.rb [hunk 2]
```diff
@@ -1,18 +1,19 @@
...
+  option "with-llvm", "(Use LLVM to compile and enable OpenMP support)"
+  depends_on "llvm" => [:build, :optional]
```
Comment (fxcoudert) [inline Formula/par2.rb]: We shouldn't depend on `llvm`, but use: `option "with-openmp", "Compile with OpenMP support"` and `needs :openmp if build.with? "openmp"` which handles it automatically. See for example how `imagemagick` does it.

**Review by fxcoudert — CHANGES_REQUESTED** [review]

**Comment by outlyer** [comment]
@fxcoudert I've added the changes you requested.

File: Formula/par2.rb [hunk 3]
```diff
@@ -11,8 +11,16 @@ class Par2 < Formula
     sha256 "7257b39640dcf1894c2329129406a573b04dd263bf3b06283dc854b5ed17cf8e" => :el_capitan
   end
 
+  option "with-openmp", "Compile with OpenMP support"
+  needs :openmp if build.with? "openmp"
+
   def install
-    system "./configure", "--prefix=#{prefix}"
+    if build.with? "openmp"
+      ENV["CXX"] = "#{Formula["llvm"].opt_bin}/clang++"
```
Comment (fxcoudert) [inline Formula/par2.rb]: No, we support OpenMP in other ways, such as using GCC as a compiler. So this formula should not hardcode anything specific to llvm.

The configure script uses `AC_OPENMP`, so OpenMP should be automatically detected and used if available.

**Comment by ilovezfs** [comment]
Going to pass on adding an `openmp` option as we generally only add it when required. Thanks for the PR nonetheless @outlyer.

**Pull request closed without merging**